### PR TITLE
Remove gdal

### LINKED ISSUE #398: gdal setup on travis sometimes fails
# Description

... maybe we can simplify the Travis setup by using a Conda environment? 

Currently I only use pywps with Conda.

# Steps to Reproduce

# Additional Information

### PR DESCRIPTION
**Pull request by cehbrecht**
# Overview

This PR removes the `gdal` dependency and replaces it with:
* [geotiff](https://pypi.org/project/geotiff/)
* [fiona](https://pypi.org/project/Fiona/)

`gdal` was used in the validator for formats: geotiff, geojson, shapefile, gml.

This PR also skips tests for Python 3.6 and CI is run now on `ubuntu-latest`.

# Related Issue / Discussion

Fix #398

# Additional Information

See also PR #636 

And [noGDAL](https://kipling.medium.com/nogdal-e5b60b114a1c) discussion.

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [ ] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [x] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

### CODE REVIEW

**Review by huard — APPROVED** [review]
Looks good! 
There's just test_ows.py that looks like a left-over. What should we do with it ?

**Comment by huard** [comment]
I think we can just remove the section you commented out. I don't think there's any added value to the feature processing going on in the test.

File: pywps/validator/complexvalidator.py [hunk 1]
```diff
@@ -55,11 +55,11 @@ def validategml(data_input, mode):
...
-        from pywps.dependencies import ogr
-        data_source = ogr.Open(data_input.file)
-        if data_source:
-            passed = (data_source.GetDriver().GetName() == "GML")
-        else:
+        try:
+            import fiona
+            data_source = fiona.open(data_input.file)
+            passed = (data_source.driver == "GML")
+        except Exception:
```
Comment (Zeitsperre) [inline pywps/validator/complexvalidator.py:62]:
```suggestion
        except (ModuleNotFoundError, ImportError):
```
We don't want to except `FileNotFoundErrrors` or deprecated API errors

File: pywps/validator/complexvalidator.py [hunk 2]
```diff
@@ -109,11 +109,11 @@ def validategpx(data_input, mode):
...
-        from pywps.dependencies import ogr
-        data_source = ogr.Open(data_input.file)
-        if data_source:
-            passed = (data_source.GetDriver().GetName() == "GPX")
-        else:
+        try:
+            import fiona
+            data_source = fiona.open(data_input.file)
+            passed = (data_source.driver == "GPX")
+        except Exception:
```
Comment (Zeitsperre) [inline pywps/validator/complexvalidator.py:116]: Same comment as above

File: pywps/validator/complexvalidator.py [hunk 3]
```diff
@@ -249,11 +249,11 @@ def validategeojson(data_input, mode):
...
-        from pywps.dependencies import ogr
-        data_source = ogr.Open(data_input.file)
-        if data_source:
-            passed = (data_source.GetDriver().GetName() == "GeoJSON")
-        else:
+        try:
+            import fiona
+            data_source = fiona.open(data_input.file)
+            passed = (data_source.driver == "GeoJSON")
+        except Exception:
```
Comment (Zeitsperre) [inline pywps/validator/complexvalidator.py:256]: Same comment as above

File: pywps/validator/complexvalidator.py [hunk 4]
```diff
@@ -317,22 +317,11 @@ def validateshapefile(data_input, mode):
...
-            data_source = ogr.Open(os.path.join(data_input.tempdir, shape_name))
-
-        if data_source:
-            passed = (data_source.GetDriver().GetName() == "ESRI Shapefile")
-        else:
+        try:
+            import fiona
+            sf = fiona.open(data_input.file)
+            passed = (sf.driver == "ESRI Shapefile")
+        except Exception:
```
Comment (Zeitsperre) [inline pywps/validator/complexvalidator.py:324]: Same comment as above

File: pywps/validator/complexvalidator.py [hunk 5]
```diff
@@ -357,10 +346,10 @@ def validategeotiff(data_input, mode):
...
 
         try:
-            from pywps.dependencies import osgeo # noqa
-            data_source = osgeo.gdal.Open(data_input.file)
-            passed = (data_source.GetDriver().ShortName == "GTiff")
-        except ImportError:
+            from geotiff import GeoTiff
+            data_source = GeoTiff(data_input.file)
+            passed = (data_source.crs_code > 0)
+        except Exception:
```
Comment (Zeitsperre) [inline pywps/validator/complexvalidator.py:352]: Same comment as above

**Review by Zeitsperre — APPROVED** [review]
Nicely done. One less project needing GDAL, one less headache.

**Pull request merged**